### ISSUE TITLE
[Test] Add RayJob + Kueue e2e tests 

### ISSUE THREAD

**Issue by andrewsykim** [ray-project/kuberay#1846]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

I'd like to add some e2e tests involving RayJob and Kueue. This ensures that any changes in Kuberay will be compatible with the latest Kueue version.

### Use case

Ensure compatibility and test coverage with RayJob and Kueue (often used for priority/gang scheduling with Ray)

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by andrewsykim** [comment]
@kevin85421 any objections?

**Comment by kevin85421** [comment]
The compatibility tests for Kueue and KubeRay are great! Do you plan to install Kueue using (1) GitHub Action YAMLs like [e2e-tests.yaml](https://github.com/ray-project/kuberay/blob/99f45c015cbb2ae5ac9acfba9191edeb3dcbaa54/.github/workflows/e2e-tests.yaml#L55-L66), or (2) the Golang files in `ray-operator/test/e2e`? If option (2) introduces significant complexity, I might prefer option (1). cc @astefanutti for additional input.

**Comment by astefanutti** [comment]
I think Kueue is lightweight enough so we could have it installed along with KubeRay in the e2e tests GH Actions workflow.

Maybe we can have a small overlay so we overwrite the default configuration to tailor it for the e2e tests, e.g., for disabling all the integrations expect but `ray.io/rayjob` and `ray.io/raycluster`.

**Comment by kevin85421** [comment]
> Are we directly using install command in [Document](https://kueue.sigs.k8s.io/docs/installation/#install-a-released-version) and add to [e2e-tests.yaml](https://github.com/ray-project/kuberay/blob/99f45c015cbb2ae5ac9acfba9191edeb3dcbaa54/.github/workflows/e2e-tests.yaml#L55-L66)?

Yes

> Do we specify the release version in the command or a YAML field?

Maybe use the latest release of Kueue (i.e. v0.8.0 for now). HDYT cc @andrewsykim @astefanutti?

**Comment by astefanutti** [comment]
I agree, using the latest stable release sounds like the best approach.

Possibly also parameterising that version in a central place, like in the Makefile, and have the e2e-tests workflow source the Kueue version from there.
